pkg/operator/ceph/pool/controller.go

```diff
@@ -288,3 +311,32 @@ func updateStatus(client client.Client, poolName types.NamespacedName, status st
 	}
 	logger.Debugf("pool %q status updated to %q", poolName, status)
 }
+
+func configureRBDStats(clusterContext *clusterd.Context, clusterName string) error {
+	cephBlockPoolList := cephv1.CephBlockPoolList{}
+	var enableStatsForPools []string
+	err := clusterContext.Client.List(context.TODO(), &cephBlockPoolList)
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve list of CephBlockPool")
+	}
+	for _, cephBlockPool := range cephBlockPoolList.Items {
+		if cephBlockPool.GetDeletionTimestamp() == nil && cephBlockPool.Spec.EnableRBDStats {
```

Comment:
Not needed at all times, but this would probably catch corner cases where pool is marked for deletion right after it was created/updated (while we are attempting to enable stats)?

Reply:
Also Refer: https://github.com/rook/rook/pull/5680#discussion_r451665347